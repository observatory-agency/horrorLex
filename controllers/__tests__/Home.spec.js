const HomeController = require('../Home');

describe('HomeController', () => {
  let home;
  beforeEach(() => {
    home = new HomeController();
  });

  describe('constructor', () => {
    it('should set its templates', () => {
      expect(home.template).toMatchSnapshot();
    });
  });

  describe('get', () => {
    it('should call render with the "get" template', () => {
      const renderMock = jest.fn();
      const resMock = { render: renderMock };
      home.get(null, resMock);
      expect(renderMock).toHaveBeenCalledWith(home.template.get);
    });
  });
});
