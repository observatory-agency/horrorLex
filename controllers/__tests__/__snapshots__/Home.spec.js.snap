// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeController constructor should set its templates 1`] = `
Object {
  "get": "home.hbs",
}
`;
